File: code/jvm/repository/src/main/java/pt/isel/repositories/messages/MessageRepository.java
package pt.isel.repositories.messages;

import pt.isel.domain.channels.Channel;
import pt.isel.domain.messages.Message;
import pt.isel.domain.users.UserInfo;
import pt.isel.repositories.Repository;

import java.time.LocalDateTime;
import java.util.List;

public interface MessageRepository extends Repository<Message> {
    Message create(String content, UserInfo user, Channel channel, LocalDateTime createdAt);

    List<Message> findAllInChannel(Channel channel, int limit, int offset);
}

File: code/jvm/repository/src/main/kotlin/pt/isel/repositories/messages/MessageRepository.kt
package pt.isel.repositories.messages

import pt.isel.domain.channels.Channel
import pt.isel.domain.messages.Message
import pt.isel.domain.users.UserInfo
import pt.isel.repositories.Repository

/** Repository interface for managing messages, extends the generic Repository */
interface MessageRepository : Repository<Message> {
    fun create(
        content: String,
        user: UserInfo,
        channel: Channel,
    ): Message

    fun findAllInChannel(
        channel: Channel,
        limit: Int = 50,
        offset: Int = 0,
    ): List<Message>
}
